src/Compilers/Core/Portable/Emit/EditAndContinue/DeletedSourceEventDefinition.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection.Metadata;
+using Microsoft.Cci;
+
+namespace Microsoft.CodeAnalysis.Emit.EditAndContinue
+{
+    internal sealed class DeletedSourceEventDefinition
+        : DeletedSourceDefinition<IEventDefinition>, IDeletedEventDefinition
+    {
+        private readonly EventDefinitionHandle _handle;
+
+        public DeletedSourceEventDefinition(IEventDefinition oldEvent, EventDefinitionHandle handle, Dictionary<ITypeDefinition, DeletedSourceTypeDefinition> typesUsedByDeletedMembers, ICustomAttribute? deletedAttribute)
```

Comment:
Can `deletedAttribute` be `null` here? Same comment applies to `DeletedSourcePropertyDefinition`